### PR TITLE
Use DuckDB to create the RepresentativePeriod structure

### LINKED ISSUE #495: Use dataframes instead of dict for RP weight matrix
### What and Why

Found this in `io.jl`.
Might already be done? @abelsiqueira  

    # TODO: Depending on the outcome of issue #294, this can be done more efficiently with DataFrames, e.g.,
    # combine(groupby(rp_mapping_df, :rep_period), :weight => sum => :weight)

### Possible Drawbacks

_No response_

### Related Issues

#294

### LINKED ISSUE #707: Refactor to create RepresentativePeriod from DuckDB

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

Atomic commit to refactor the RepresentativePeriod structure to be created using DuckDB instead of table_tree

## List of related issues or pull requests

Closes #707 
Closes #495 

## Collaboration confirmation

As a contributor I confirm

-   [x] I read and followed the instructions in README.dev.md
-   [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
-   [x] Tests are passing
-   [x] Lint is passing

### CODE REVIEW

**Comment by abelsiqueira** [comment]
It loses a bit of efficiency (< 1.5%). Is it within acceptable parameters?

**Comment by datejada** [comment]
> It loses a bit of efficiency (< 1.5%). Is it within acceptable parameters?

I would say yes since most of the time is spent in other parts of the code. But, we need to keep an eye on it.

File: src/io.jl [hunk 1]
```diff
@@ -122,20 +122,23 @@ The details of these structures are:
...
+    weights =
+        DBInterface.execute(
+            connection,
+            "SELECT rep_period, SUM(weight) AS weight
+                FROM rep_periods_mapping
+                GROUP BY rep_period
+                ORDER BY rep_period",
+        ) |>
+        DataFrame |>
+        df -> df.weight
```
Comment (suvayu) [inline src/io.jl:137]: If the intention is to get back a row for each representative period, where the weight is like a "count" of all periods that is associated to that representative period, then this looks correct.

**Review by suvayu — APPROVED** [review]

File: src/io.jl (hunk 1 above)
Comment (datejada) [reply src/io.jl:137]: Thanks! Great, that is the intention 😄

**Pull request merged**